lib/lure/command/updateDependencies.go

```diff
@@ -0,0 +1,217 @@
+package command
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+
+	"github.com/coveooss/lure/lib/lure"
+	"github.com/coveooss/lure/lib/lure/log"
+	"github.com/coveooss/lure/lib/lure/project"
+	p "github.com/coveooss/lure/lib/lure/provider"
+	"github.com/coveooss/lure/lib/lure/versionManager"
+	"github.com/coveooss/lure/lib/lure/versionManager/mvn"
+	"github.com/coveooss/lure/lib/lure/versionManager/npm"
+
+	"github.com/vsekhar/govtil/guid"
+)
+
+// This part interesting
+// https://github.com/golang/go/blob/1441f76938bf61a2c8c2ed1a65082ddde0319633/src/cmd/go/vcs.go
+
+func appendIfMissing(modules []versionManager.ModuleVersion, modulesToAdd []versionManager.ModuleVersion) []versionManager.ModuleVersion {
+	for _, moduleToAdd := range modulesToAdd {
+		exist := false
+		for _, module := range modules {
+			if (moduleToAdd.Name != "" && module.Name == moduleToAdd.Name) || reflect.DeepEqual(module, moduleToAdd) {
+				exist = true
+				break
+			}
+		}
+		if !exist {
+			modules = append(modules, moduleToAdd)
+		}
+	}
+
+	return modules
+}
+
+func CheckForUpdatesJobCommand(project project.Project, sourceControl sourceControl, provider provider, args map[string]string) error {
+	return checkForUpdatesJob(project, sourceControl, provider, args["commitMessage"])
+}
+
+func checkForUpdatesJob(project project.Project, sourceControl sourceControl, provider provider, commitMessage string) error {
+	log.Logger.Infof("switching to default branch: %s", project.DefaultBranch)
+	if _, err := sourceControl.Update(project.DefaultBranch); err != nil {
+		return fmt.Errorf("Error: \"Could not switch to branch %s\" %s", project.DefaultBranch, err)
+	}
+
+	modulesToUpdate := make([]versionManager.ModuleVersion, 0, 0)
+
+	if project.SkipPackageManager == nil || project.SkipPackageManager["npm"] != true {
+		modulesToUpdate = appendIfMissing(modulesToUpdate, npm.NpmOutdated(sourceControl.WorkingPath()))
+	}
+
+	if project.SkipPackageManager == nil || project.SkipPackageManager["mvn"] != true {
+		err, modulesToAdd := mvn.MvnOutdated(sourceControl.WorkingPath())
+		if err != nil {
+			return err
+		}
+		modulesToUpdate = appendIfMissing(modulesToUpdate, modulesToAdd)
+	}
+
+	log.Logger.Infof("Modules to update : %q", modulesToUpdate)
+
+	ignoreDeclinedPRs := os.Getenv("IGNORE_DECLINED_PR") == "1"
+	pullRequests, err := provider.GetPullRequests(project.Owner, project.Name, ignoreDeclinedPRs)
+	if err != nil {
+		return err
+	}
+
+	for _, moduleToUpdate := range modulesToUpdate {
+		updateModule(moduleToUpdate, project, sourceControl, provider, pullRequests, commitMessage)
+	}
+
+	err = closeOldBranchesWithoutOpenPR(project, sourceControl, provider)
+	if err != nil {
+		return err
+	}
+
+	log.Logger.Infof("`[Check for updates done.")
```

Comment:
🤔 It might be my Go noobyness, but this line looks weird.

Reply:
yeah no, that's def. weird. I'll look into it thanks.